ts/util/legend.ts

```diff
@@ -0,0 +1,159 @@
+    title.innerHTML = "Legend";
+    const spacer = document.createElement("div");
+    spacer.className = "flex-auto";
+    const icon = makeIcon("xmark");
+    icon.classList.add(
+      "fa-solid",
+      "fa-xmark",
+      "text-grey-200",
+      "hover:text-green-200",
+      "text-24",
+      "m-2",
+      "mx-4",
+      "z-50",
+      "relative",
+      "cursor-pointer"
+    );
+    icon.addEventListener("click", () => this.container.remove());
+    header.appendChild(title);
+    header.appendChild(spacer);
+    header.appendChild(icon);
+    // grid of legend labels
+    const body = document.createElement("div");
+    body.className = "m-4 grid grid-rows-4 grid-flow-col gap-2";
+    for (const qcStatus of Object.values(qcStatuses)) {
+      const labelContainer = document.createElement("div");
+      labelContainer.className =
+        "flex items-center space-x-2 bg-grey-100 rounded-md font-inter font-medium p-2 text-12";
+      const icon = makeIcon(qcStatus.icon);
+      icon.classList.add("text-black");
+      const label = document.createElement("span");
+      label.innerHTML = qcStatus.label;
+      labelContainer.appendChild(icon);
+      labelContainer.appendChild(label);
+      body.appendChild(labelContainer);
+    }
+
+    // centering window, setTimeout is required for offsetHeight and offsetWidth to load
+    // TODO: fix flash introduced by timeout
+    window.setTimeout(() => {
+      this.container.style.top =
+        (window.innerHeight - this.container.offsetHeight) / 2 + "px";
+      this.container.style.left =
+        (window.innerWidth - this.container.offsetWidth) / 2 + "px";
+    }, 0);
+
+    this.container.append(header);
+    this.container.append(body);
+    // add dragging functionality
+    this.makeDraggable(header);
+  }
+
+  private makeDraggable(windowContainerTop: HTMLElement) {
+    //state
+    let diffX = 0,
+      diffY = 0,
+      previousPosX = 0,
+      previousPosY = 0;
+
+    const dragMouseDown = (e: MouseEvent) => {
+      // Prevent any default action on this element (you can remove if you need this element to perform its default action)
+      e.preventDefault();
+      // Get the mouse cursor position and set the initial previous positions to begin
+      previousPosX = e.clientX;
+      previousPosY = e.clientY;
+      // When the mouse is let go, call the closing event
+      document.onmouseup = closeDragElement;
+      // call a function whenever the cursor moves
+      document.onmousemove = elementDrag;
+    };
+
+    const elementDrag = (e: MouseEvent) => {
+      // Prevent any default action on this element (you can remove if you need this element to perform its default action)
+      e.preventDefault();
+
+      // disable window moving if the cursor is out of window
+      if (
+        e.clientX < 0 ||
+        e.clientX > window.innerWidth ||
+        e.clientY < 0 ||
+        e.clientY > window.innerHeight
+      ) {
+        return;
+      }
```

Comment:
Remove this and it'll have the behaviour you wanted - it'll be bound to the window, but will still move on any axis that you're in bounds of

Reply:
I prefer just disabling moving the window entirely when the cursor is OOB for now. With this block removed the window will move on both axes, including the one OOB.

Ideally, it only moves in the axis thats in bounds. (e.g if my cursor is OOB on the left, then the window still moves on the y-axis but stays locked on the x-axis until the cursor goes back into the window). [This demo](https://www.columbia.edu/~njn2118/journal/2019/4/26.htmll) gets pretty close to what I'm looking for, but they seem to use some slightly different math that I couldn't crack before I made the PR :(

Reply:
The clamp functions prevent the window from going out of bounds, so it will only move on an axis that is in bounds if this is removed. I have tried and verified this